include/ensmallen_bits/lmcma/lmcma.hpp

```diff
@@ -0,0 +1,83 @@
+//
+// Created by o on 07.04.21.
+//
+
+#ifndef ENSMALLEN_LMCMA_LMCMA_HPP
+#define ENSMALLEN_LMCMA_LMCMA_HPP
+
+
+namespace ens {
+
+
+class LMCMA
+{
+  public:
+
+    LMCMA(std::size_t N_dim);
+
+    template<typename SeparableFunctionType,
+             typename MatType,
+             typename... CallbackTypes>
+    typename MatType::elem_type optimize(SeparableFunctionType& f,
+                                         MatType& z,
+                                         float sigma,         // TODO: remove from here
+                                         std::size_t n_iter,  // TODO: Remove from here
+                                         CallbackTypes&&... callbacks);
+  private:
+
+    template <typename BaseMatType>
+    size_t update(std::size_t t,
+                         const BaseMatType& p,
+                         BaseMatType& P,
+                         BaseMatType& V,
+                         arma::umat& L,
+                         arma::umat& J);
+
+
+      template <typename BaseMatType>
+      void reconstruct(const BaseMatType& P,
+                       const BaseMatType& V,
+                       const arma::umat& J,    /* TODO: why umat? */
+                       const std::size_t n_updates,
+                       BaseMatType& z);
```

Comment:
```BaseMatType``` is only used to convert Row/Col ==> Mat type inside our ```Optimize()``` function. For others, the convention is to just write ```MatType```. See for ex: [cne.hpp](https://github.com/mlpack/ensmallen/blob/1b6c1d93f09adc2313c67827355e6f9c9f958d3c/include/ensmallen_bits/cne/cne.hpp#L152). 

Also, function names should start with capital followed by small case letters.

```suggestion
      template <typename MatType>
      void Reconstruct(const MatType& P,
                       const MatType& V,
                       const arma::umat& J,    /* TODO: why umat? */
                       const std::size_t n_updates,
                       MatType& z);
```

Reply:
Same for other functions.

Reply:
Makes sence